operator/pkg/central/values/translation/translation.go

```diff
@@ -204,11 +204,17 @@ func getCentralDBComponentValues(c *platform.CentralDBSpec) *translation.ValuesB
 	cv := translation.NewValuesBuilder()
 	cv.SetBoolValue("enabled", true)
 
-	if c.ConnectionStringOverride != nil && c.PasswordSecret == nil {
-		cv.SetError(errors.New("if connection string override is set, then a password secret must also be set"))
-	}
-
 	if c.ConnectionStringOverride != nil {
+		if c.GetPersistence() != nil {
+			cv.SetError(errors.New("if a connection string is provided, no persistence settings must be supplied"))
+		}
+
+		// TODO: there are other settings which are ignored in external mode - should we error if those are set, too?
```

Comment:
+1 for erroring loud and clear on dubious configurations

Reply:
Yeah, I think now I remember the problem, but that problem is even bigger.

The OpenShift console has the very unfortunate property of importing OpenAPI field defaults into the form _and then acting as if the user has set those field values_. See:

https://user-images.githubusercontent.com/2822367/198082234-7d0a986b-e903-44c6-828d-f1c233875a6e.mov


Currently there's no issue because Gorman didn't introduce a default value for the PVC name for the DB, but, arguably, there should be one, as we also have a `stackrox-db` default for the central persistence, and setting a default for a field is really nothing very unreasonable.

The bigger problem now is that setting a default for any of the fields means a user using OpenShift console can no longer disable central DB. OpenShift console simply does not respect the fact that (non-)nullness of fields in the CR might carry any meaning, it uses a lossy round-trip between the two representations. This, by the way, also applies in the other direction: with the existing CR structure, it is impossible to specify "use central DB, but with default settings" in the form view in OpenShift console, you have to set any of the fields (such as explicitly typing in the default `central-db` pvc name) or use YAML view.

@porridge I know you had some discussion with Gorman on the original PR re: an explicit boolean field vs the mere presence of another field (this was about external, not the DB as a whole). Putting the aspect of bool vs enum aside: what is your opinion on (a) avoiding explicit enabled/disabled fields and (b) erroring on inconsistent combinations of set fields, in light of these OpenShift console form view realities?

Reply:
Oh, wow. This is a game-changer. :disappointed:
I guess this means that mere presence of a struct can no longer have any meaning.
This in turn means `CentralDBEnabled()` needs to look at a new explicit toggle (I would very much prefer an enum on the `CentralComponentSpec` level rather than bools) rather than at the nilness of `.DB` :thinking:

Please also mention in a (non-user visible) comment somewhere why we don't look at struct nilness. This is probably not obvious for a non-openshift developer.